### DISCUSSION TITLE
how do you edit/develop these controls? (defaults is not recognized, theme json)

### DISCUSSION THREAD

**Discussion by DaFlori** [pnp/powerapps-designtoolkit#15] [General]
I had a quick peek into the components within the imported component library. 
The components showing me errors "defaults is not recognized" for every property that tries to use this "defaults"-object (theming?), as in 
property: primaryColorHex
value: defaults.primaryColorHex

So how do you develop these components i. e. how to provide this "defaults"-object during editing a component?

**Comment by geoffroff** [comment]
The defaults variable should be created in the App.OnStart or <screen>.OnVisible action properties. Within the screens section in the component library, you can see an example in SetupPage.OnVisible. The Set() command will create the defaults table and Patch() calculated columns. When you then drop controls onto screen you will not receive errors.
See [video](https://www.youtube.com/watch?v=kkGjJBQoI3I) that might explain further.
Depending on whether this library will continue to grow and develop, if you need 'changes' then you might be best to create your own MD library components. This way, you could still update the MD solution from github and not destroy any change you might have made. Having said that it would be worth reading through the https://m3.material.io/ to follow the MD pathway.
